Orange/base.py

```diff
@@ -63,15 +63,29 @@ def __call__(self, data):
         model.name = self.name
         model.original_domain = origdomain
         return model
-
+    
     def preprocess(self, data):
         """
         Apply the `preprocessors` to the data.
         """
-        for pp in self.preprocessors:
+        for pp in self.active_preprocessors:
             data = pp(data)
         return data
 
+    @property
+    def active_preprocessors(self):
+        if self.preprocessors is not None:
+            pps = []
+            if isinstance(self.preprocessors, Iterable):
+                pps += list(self.preprocessors)
+            else:
+                pps += [self.preprocessors]
```

Comment:
I thought about this too, but we do similar things in other places, and Python does it too, like in, for instance, the second argument of `isinstance`. Practicality beats purity.